### ISSUE TITLE
ooniprobe: v3.5.1 (21.02) does not start on x86

### ISSUE THREAD

**Issue by ValdikSS** [openwrt/packages#18219]
Maintainer: @ja-pa, @neheb
Environment: 21.02 x86 generic

Description:
ooniprobe main executable from ooniprobe package does not start, at least on x86 platform (have not tested other platforms).

```
root@OpenWrt:~# ooniprobe 
panic: qtls.ClientHelloInfo doesn't match

goroutine 1 [running]:
github.com/marten-seemann/qtls-go1-15.init.0()
        github.com/marten-seemann/qtls-go1-15@v0.1.1/unsafe.go:20 +0x191

root@OpenWrt:~# apitool
panic: qtls.ClientHelloInfo doesn't match

goroutine 1 [running]:
github.com/marten-seemann/qtls-go1-15.init.0()
        github.com/marten-seemann/qtls-go1-15@v0.1.1/unsafe.go:20 +0x191

root@OpenWrt:~# miniooni
panic: qtls.ClientHelloInfo doesn't match

goroutine 1 [running]:
github.com/marten-seemann/qtls-go1-15.init.0()
        github.com/marten-seemann/qtls-go1-15@v0.1.1/unsafe.go:20 +0x191
```

Any argument result in the panic above.

**Comment by jamesmacwhite** [comment]
At a guess, I would suspect a Go version incompatibility. Similar errors have been seen before with AdGuardHome.

The fact the stacktrace mentions Go 1.15 suggests this, but I'm no expert. OpenWrt 21.02 builds with Go 1.17 for reference.

I've never used the package in question, but the latest release is v3.14.1 and it seems to be built against Go 1.17, so bumping the package version might just fix the issue.

**Comment by ValdikSS** [comment]
>However, you'd have to confirm the latest version can build on Go 1.18, as master is using that version now, before backporting.

I have close-to-zero experience in Golang.

**Comment by jamesmacwhite** [comment]
@ValdikSS Good new. Building the latest version for 21.02.2 it 

```
root@linksys-wrt3200acm:~# ooniprobe version
3.14.1

root@linksys-wrt3200acm:~# apitool --help
Usage of apitool:
  -input string
        Input of the measurement
  -mode string
        One of: check, meta, raw
  -report-id string
        Report ID of the measurement
  -v    Enable verbose mode

root@linksys-wrt3200acm:~# miniooni --version
3.14.1
```

So we should be able to backport the latest version down to 21.02 to fix it. It builds for both Go 1.17 and 1.18

**Comment by jamesmacwhite** [comment]
This should be available in 21.02 now.

**Comment by ValdikSS** [comment]
It works, thanks a lot!
